panther_analysis_tool/util.py

```diff
@@ -252,3 +253,31 @@ def log_and_write_to_file(msgs: List[str], filename: TextIO) -> None:
     for msg in msgs:
         filename.write(msg + "\n")
         logging.info(msg)
+
+
+def lookup_analysis_id(analysis_spec: Any) -> str:
+    analysis_type = analysis_spec.get("AnalysisType")
+    analysis_id = UNKNOWN_ANALYSIS_ID
+    if analysis_type == AnalysisTypes.DATA_MODEL:
+        analysis_id = analysis_spec["DataModelID"]
+    elif analysis_type == AnalysisTypes.GLOBAL:
+        analysis_id = analysis_spec["GlobalID"]
+    elif analysis_type == AnalysisTypes.LOOKUP_TABLE:
+        analysis_id = analysis_spec["LookupName"]
+    elif analysis_type == AnalysisTypes.PACK:
+        analysis_id = analysis_spec["PackID"]
+    elif analysis_type == AnalysisTypes.POLICY:
+        analysis_id = analysis_spec["PolicyID"]
+    elif analysis_type == AnalysisTypes.SCHEDULED_QUERY:
+        analysis_id = analysis_spec["QueryName"]
+    elif analysis_type == AnalysisTypes.SAVED_QUERY:
+        analysis_id = analysis_spec["QueryName"]
+    elif analysis_type == AnalysisTypes.SIGNAL:
+        analysis_id = analysis_spec["SignalID"]
+    elif analysis_type in [
+        AnalysisTypes.RULE,
+        AnalysisTypes.SCHEDULED_RULE,
+        AnalysisTypes.CORRELATION_RULE,
+    ]:
```

Comment:
oh, also... i realize that this is how the function already behaved, so i don't wanna derail anything here... but i assume that we don't actually do anything with `UNKNOWN_ANALYSIS_ID`.

in the future, it might be better to add:
```
else:
    raise UnknownAnalyasisID
```

and then callers can handle that exception rather than check strings.